File: 2024/intermediate/01-find-median/solutions/main.py
K = int(input())
arraynumbers = []
for i in range(K):
    arraynumbers.append(int(input()))
arraynumbers.sort()
if(K % 2 == 0 ):
    median = (arraynumbers[K // 2] + arraynumbers[(K // 2) - 1]) / 2.0
    if median.is_integer():
        print(int(median))
    else:
        print(median)
else:
    print(arraynumbers[K // 2])

File: 2024/intermediate/01-find-median/solutions/main.cs
using System;

internal class Program
{
    static void Main(string[] args)
    {
        int K = int.Parse(Console.ReadLine());
        double[] arraynumbers = new double[K];
        for (int i = 0; i < K; i++){
            arraynumbers[i] = int.Parse(Console.ReadLine());
        }
        Array.Sort(arraynumbers);
        if (K % 2 == 0)
        {
            Console.WriteLine((arraynumbers[K / 2] + arraynumbers[(K / 2)-1])/2);
        }
        else
        {
            Console.WriteLine(arraynumbers[K / 2]);
        }
       
    }
}
